src/OrleansCodeGenerator/RoslynCodeGenerator.cs

```diff
@@ -429,10 +429,13 @@ private static GeneratedSyntax GenerateForAssemblies(List<Assembly> assemblies,
             if (typeInfo.IsSerializable)
                 RecordType(type, module, targetAssembly, includedTypes);
 
-            Type grainStateType;
+            Type typeToRecord;
             // If a type extends Grain<T>, add T to the list of candidates for serialization
-            if (IsPersistentGrain(typeInfo, out grainStateType))
-                RecordType(grainStateType, module, targetAssembly, includedTypes);
+            if (IsPersistentGrain(typeInfo, out typeToRecord))
+                RecordType(typeToRecord, module, targetAssembly, includedTypes);
+
+            if (IsAsyncObserver(typeInfo, out typeToRecord))
+                RecordType(typeToRecord, module, targetAssembly, includedTypes);
```

Comment:
You mean record ALL types? But that would mean ANY type defined in the assembly, including ones that were never intended to be serialized. Would not that be a problem? Would we always succeed, on any type?